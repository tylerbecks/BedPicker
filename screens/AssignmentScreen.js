import React from 'react';
import { StyleSheet, View } from 'react-native';
import BedSection from '../components/BedSection';
import Assigner from '../components/Assigner';

const tahoeBeds = [
  { type: 'twin', name: 'upstairs twin', guests: [] },
  { type: 'queen', name: 'master', guests: [] },
  { type: 'twin', name: 'upstairs twin', guests: [] },
  { type: 'full', name: 'upstairs full', guests: [] },
  { type: 'full', name: 'couch', guests: [] }
];

//  bed: {
//    type: string
//    name: string
//    guests: [string]
//  }

export default class AssignmentScreen extends React.Component {
  state = {
    assignment: null
  };

  componentDidMount() {
    const selectedGuests = this.props.navigation.getParam('selectedGuests', []);
    const assigner = new Assigner(selectedGuests, tahoeBeds);
    this.setState({ assignment: assigner.assign() });
  }

  render() {
    return (
      <View style={styles.container}>
        {this.state.assignment &&
          this.state.assignment.map(({ type, name, guests }, index) => (
            <BedSection key={index} type={type} name={name} guests={guests} />
          ))}
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff'
  }
});
