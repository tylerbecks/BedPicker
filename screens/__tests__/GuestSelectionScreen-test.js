import 'react-native';
import React from 'react';
import AssignmentScreen from '../GuestSelectionScreen';
import renderer from 'react-test-renderer';

it('renders', () => {
  expect(renderer.create(<AssignmentScreen />).toJSON()).toMatchSnapshot();
});
