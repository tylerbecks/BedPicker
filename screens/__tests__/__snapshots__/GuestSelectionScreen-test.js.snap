// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
/>
`;
